src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -0,0 +1,720 @@
+
+            # If we're still here, we must not have a real error report
+            logOutput = 'Could not find jobReport\n'
+            # But we don't know exactly the condor id, so it will append
+            # the last lines of the latest condor log in cache_dir
+            # TODO: Alan - how about we look at the .out or .err file instead of the .log?
+            genLogPath = os.path.join(job['cache_dir'], 'condor.*.*.log')
+            logPaths = glob.glob(genLogPath)
+            errLog = None
+            if len(logPaths):
+                errLog = max(logPaths, key=lambda path: os.stat(path).st_mtime)
+            if errLog is not None and os.path.isfile(errLog):
+                logTail = BasicAlgos.tail(errLog, 50)
+                logOutput += 'Adding end of condor.log to error message:\n'
+                logOutput += '\n'.join(logTail)
+
+            condorReport = Report()
+            if not os.path.isdir(job['cache_dir']):
+                msg = "Serious Error in Completing condor job with id %s!\n" % job.get('id', 'unknown')
+                msg += "Could not find jobCache directory - directory deleted under job: %s\n" % job['cache_dir']
+                msg += "Creating artificial cache_dir for failed job report\n"
+                logging.error(msg)
+                os.makedirs(job['cache_dir'])
+                logOutput += msg
+                condorReport.addError("NoJobReport", 99304, "NoCacheDir", logOutput)
+                condorReport.save(filename=reportName)
+                continue
+
+            condorReport.addError("NoJobReport", 99303, "NoJobReport", logOutput)
+            if os.path.isfile(reportName):
+                # Then we have a file already there.  It should be zero size due
+                # to the if statements above, but we should remove it.
+                if os.path.getsize(reportName) > 0:
+                    # This should never happen.  If it does, ignore it
+                    msg = "Critical strange problem.  FWJR changed size while being processed."
+                    logging.error(msg)
+                else:
+                    try:
+                        os.remove(reportName)
+                        condorReport.save(filename=reportName)
+                    except:
+                        logging.error("Cannot remove and replace empty report %s", reportName)
+                        logging.error("Report continuing without error!")
+            else:
+                condorReport.save(filename=reportName)
+
+            # Debug message to end loop
+            logging.debug("No returning job report for job %i", job['id'])
+
+        return
+
+    def updateSiteInformation(self, jobs, siteName, excludeSite):
+        """
+        _updateSiteInformation_
+
+        Modify condor classAd for all Idle jobs for a site if it has gone Down, Draining or Aborted.
+        Kill all jobs if the site is the only site for the job.
```

Comment:
It only kills the jobs when all the desired sites are in drain status since in that case jobs can't be running any sites.
Other cases, it just modifies classAd for Desided_sites list to remove the draining site.
I am not completely sure why we keep the running jobs as it is. (I think we hope it might run if not will at least and be failed/comes back) But maybe it is better to kill those as well.